locations/spiders/lidl_pl.py

```diff
@@ -9,3 +11,38 @@ class LidlPLSpider(LidlATSpider):
     dataset_name = "Filialdaten-PL/Filialdaten-PL"
     key = "AnZ7UrM33kcHeNxFJsJ6McC4-iAx6Mv55FfsAzmlImV6eJ1n6OX4zfhe2rsut6CD"
     days = DAYS_PL
+
+    def parse_item(self, item, feature, **kwargs):
+        item["name"] = feature["ShownStoreName"]
+
+        # Takes care of the case where ul. is in street address and may or may not be in name
+        if "ul." in item["street_address"]:
+            ul_index = item["street_address"].index("ul.")
+            if item["street_address"] in item["name"]:
+                item["name"] = item["name"].replace(item["street_address"], "").strip()
+            elif item["street_address"][ul_index + 3 :].strip() in item["name"]:
+                item["name"] = item["name"].replace(item["street_address"][ul_index + 3 :].strip(), "").strip()
+        # Takes care of the case where ul. is not in street address and may or may not be in name
+        else:
+            if "ul. " + item["street_address"] in item["name"]:
+                item["name"] = item["name"].replace("ul. " + item["street_address"], "").strip()
+            elif item["street_address"] in item["name"]:
+                item["name"] = item["name"].replace(item["street_address"], "").strip()
+
+        # Takes care of the case where city is in name
+        if item["city"] + "," in item["name"]:
+            item["name"] = item["name"].replace(item["city"] + ",", "").strip()
+        elif item["city"] in item["name"]:
+            item["name"] = item["name"].replace(item["city"], "").strip()
+
+        item["name"] = item["name"].strip()
+
```

Comment:
IMO, it's not worth it. There is no reasonable branch name, so drop it/don't collect it.

```suggestion
```

Reply:
This may be the same for the other Lidl spiders, I've not looked.